test/e2e/transformations/xmltojson/main.go

```diff
@@ -0,0 +1,158 @@
+*/
+
+package xmltmtojson
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/bridges"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+)
+
+var transAPIVersion = schema.GroupVersion{
+	Group:   "flow.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	transformationKind     = "XMLToJSONTransformation"
+	transformationResource = "xmltojsontransformations"
+
+	expectedResponseEvent = "{\"string\":\"\\u003cnote\\u003e\\u003cto\\u003eTove\\u003c/to\\u003e\\u003cfrom\\u003eJani\\u003c/from\\u003e\\u003cheading\\u003eReminder\\u003c/heading\\u003e\\u003cbody\\u003eDont forget me this weekend\\u003c/body\\u003e\\u003c/note\\u003e\"}"
+)
+
+// createTransformation creates an AWSKinesis object initialized with the given options.
+func createTransformation(trnsClient dynamic.ResourceInterface, namespace, namePrefix string, dest *duckv1.Destination) (*unstructured.Unstructured, error) {
+	trns := &unstructured.Unstructured{}
+	trns.SetAPIVersion(transAPIVersion.String())
+	trns.SetKind(transformationKind)
+	trns.SetNamespace(namespace)
+	trns.SetGenerateName(namePrefix)
+
+	if dest != nil {
+		if err := unstructured.SetNestedMap(trns.Object, ducktypes.DestinationToMap(dest), "spec", "sink"); err != nil {
+			framework.FailfWithOffset(2, "Failed to set spec.sink field: %s", err)
+		}
+	}
+
+	return trnsClient.Create(context.Background(), trns, metav1.CreateOptions{})
+}
+
+// readReceivedEvents returns a function that reads CloudEvents received by the
+// event-display application and stores the result as the value of the given
+// `receivedEvents` variable.
+// The returned function signature satisfies the contract expected by
+// gomega.Eventually: no argument and one or more return values.
+func readReceivedEvents(c clientset.Interface, namespace, eventDisplayDeplName string,
+	receivedEvents *[]cloudevents.Event) func() []cloudevents.Event {
+
+	return func() []cloudevents.Event {
+		ev := bridges.ReceivedEventDisplayEvents(
+			apps.GetLogs(c, namespace, eventDisplayDeplName),
+		)
+		*receivedEvents = ev
+		return ev
+	}
+}
+
+var _ = Describe("XMLToJSON Transformation", func() {
+	f := framework.New("xmltojsontransformation")
+	var ns string
+	var sink *duckv1.Destination
+	var trnsClient dynamic.ResourceInterface
+	var trans *unstructured.Unstructured
+	var err error
```

Comment:
This `err` variable seems to only be used in one very specific `By`, and not across spec blocks, so it could be moved there instead of at the top scope.